django/applications/catmaid/static/js/widgets/stack-viewer-grid.js

```diff
@@ -369,6 +369,54 @@
     scaleBarCbLabel.appendChild(scaleBarCb);
     this.controlsContainer.appendChild(scaleBarCbLabel);
 
+    var zoomInput = document.createElement('input');
+    zoomInput.setAttribute('type', 'text');
+    zoomInput.setAttribute('pattern', '^\-?\d?\.?\d*$');
+    zoomInput.setAttribute('size', '5');
+    // zoomInput.setAttribute('type', 'number');
+    // zoomInput.setAttribute('step', '0.1');
+    // zoomInput.setAttribute('max', '5');
+    // zoomInput.setAttribute('min', '-2');
```

Comment:
Fair enough, just wanted to see what you thought of using the text type. The number type (which automatically makes a spinner which is hard to remove) or a slider encourages users to make many small increments, which is fine when you're only redrawing 1 viewer, but really kills performance when you're redrawing as many as 15. I found that when using the spinner, it would redraw fine for the first two increments but then stop until the user either pressed enter or focused the mini stack viewers to force a refresh.